[MLIR][NVVM] Update mbarrier.test.wait Op

durga4github opened the pull request:
This patch extends mbarrier.test_wait to support scope,
semantics, and phase-parity, completing the updates for
this Op up to Blackwell. Corresponding lit tests are added
to verify the lowering.

grypp commented on mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td:
```diff
@@ -980,13 +980,10 @@ def NVVM_MBarrierTryWaitParityOp : NVVM_PTXBuilder_Op<"mbarrier.try_wait.parity"
-  Results<(outs I1:$res)>,
-  Arguments<(ins AnyTypeOf<[LLVM_PointerGeneric, LLVM_PointerShared]>:$addr,
-                 I64:$state)> {
+def NVVM_MBarrierTestWaitOp : NVVM_Op<"mbarrier.test_wait"> {
```
MLIR slightly prefers "_" on the OP name and "." is only used to separate the dialect name. We have wrong name convention in NVVM. We mix "_" and ".". 

I'd like to change all the names to "_", and it's better to do it in single PR. 

Let's not change the name here

grypp reviewed (approved):
Looks great. But let's not change the name of the OP

durga4github committed:
[MLIR][NVVM] Update mbarrier.test.wait Op

This patch extends the mbarrier.test.wait Op to
support scope, semantics and phase-parity. This
completes updates to the test_wait up-to Blackwell.
lit tests are added to verify the lowering.

Signed-off-by: Durgadoss R <durgadossr@nvidia.com>
